canonical/kfp-operators
# refactor: use k8s_service_info lib instead of SDI

Use the k8s_service_info for receiving the MLMD GRPC Service info instead of using the SDI, as it will stop being supported soon.
This commit also ensures that mlmd runs with trust=True in the integration tests.

Fixes #413

NOTE for reviewers:

* I have added a component for wrapping the k8s service info requirer because we need to:
1. Have a way to return a status based on the relation conditions
2. Because there is no other way to do logic that prevents errors in the init method of the charm. For instance, it would be weird to have a try/catch in the init method for ensuring the required relation is established.

* Also note that this component could be placed in chisme, BUT that requires us to put charm libraries in the package, which is under discussion by the team.

* Depends on https://github.com/canonical/envoy-operator/pull/87

#### Testing instructions

1. Deploy the charm in this branch
2. Deploy `mlmd` from `channel latest/edge` with `trust`
3. Relate both charms
4. Verify both charms go to active and idle

##### Test upgrade path
1. Deploy `kfp-metadata-writer` from `channel 2.0/stable` with `trust`
2. Run `juju refresh` to the charm in this branch
3. Observe the charm going to Blocked(because it is missing a relation with mlmd) with the messsage: `[grpc] Missing relation with a k8s service info provider. Please add the missing relation`
